doc/platform_api/CMIS_and_C-CMIS_support_for_ZR.md

```diff
@@ -0,0 +1,1566 @@
+            ||                              /\
+            \/                              ||
+       ---------------------------------------------
+      |                   xcvrd                     |
+       ---------------------------------------------
+                    ||            /\
+                    \/            ||
+              ---------------------------
+             |    High level functions   |
+              ---------------------------
+                    ||            /\
+                    \/            ||
+                 ---------------------
+                | Encode        Decode |
+                 ---------------------
+                    ||            /\
+                    \/            ||
+                ------------------------
+               | write_reg     read_reg |
+                ------------------------               
+                    ||            /\
+                    \/            ||
+                 ---------------------
+                |   Module registers  |
+                 ---------------------
+```      
+
+##### 1.1.1 Upon plug-in of a module 
+When there is plug-in event of a module on a port, the xcvrd detects the presence state changes from "absent" to "present" on that port. xcvrd reacts to this event by pushing the config_DB on that port to the module. More specifically for ZR module, the low power mode (lpmode), the configured frequency, the configured TX power and the loopback mode are the four most important settings to decide whether the module will be turned up, and with what settings if the module is to be turned up. xcvrd also constantly polls the state of the module and update its memory. When a CLI show command queries the module state or other features, the state_DB will be updated with the information in xcvrd.
+
+##### 1.1.2 Issuing a show CLI command
+When a user issues a show CLI command, state_DB will read from xcvrd and update its fields with information from xcvrd.
+
+##### 1.1.3 Issuing a config CLI command
+When a user issues a config CLI command, config_DB will be updated, and send new settings to xcvrd. xcvrd will further push the settings into the module registers. 
+
+### 2. State_DB, Config_DB, show/config transceiver CLI definitions:
+
+#### 2.1 State_DB Schema ####
+
+New Transceiver info table and transceiver DOM sensor table adapted to 400G-ZR modules.
+
+##### 2.1.1 Transceiver info Table #####
+
+    ; Defines Transceiver information for a port
+    key                          = TRANSCEIVER_INFO|ifname          ; information for module on port
+    ; field                      = value
+    type                         = 1*255VCHAR                       ; module type full name(QSFP-DD, OSFP, etc)
+    type_abbrv_name              = 1*255VCHAR                       ; module type abbreviated name(QSFP-DD, OSFP, etc)
+    module_media_type            = 1*255VCHAR                       ; module media interface ID
```

Comment:
In today's cmis.py: get_transceiver_info() code, I didn't see module_media_type is getting filled. I only saw specification_compliance is getting filled with  get_module_media_type(). 
https://github.com/sonic-net/sonic-platform-common/blob/86bab38c723eb4ebbfa16feed66344d1b3ffd46e/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L166
```
        xcvr_info['specification_compliance'] = self.get_module_media_type()
```
Not sure if this is a duplicated field. Among 'specification_compliance' and 'module_media_type', shall we only keep one of them? Or they serve different purposes?

Reply:
I verified type_abbrv_name and module_media_type are not in state_db TRANSCEIVER_INFO table. I removed both items.